tests/test_agent.py

```diff
@@ -636,6 +636,41 @@ def validate_output(ctx: RunContext[None], o: Any) -> Any:
     assert got_tool_call_name == snapshot('final_result_Bar')
 
 
+def test_output_type_generic_class_name_sanitization(create_module: Callable[[str], Any]):
+    """Test that generic class names with brackets are properly sanitized."""
+    module_code = '''
```

Comment:
We can inline this as regular code -- we're only using a stringified module above because we're injecting code `{union_code}`.